src/commands/Quote.ts

```diff
@@ -0,0 +1,207 @@
+                  else if(this.findType[0] == '"' ){
+                    let word = ""
+                    for(let i = 1; i < this.findType.length-1; i = i + 1){
+                      word = word + this.findType[i]
+                    }
+                    let match = 0;
+                    for(let i = 0; i < Object.keys(userArray[value]).length; i = i + 1){
+                      let text = userArray[value][i].quote
+                      if(text.indexOf(word) > -1){
+                        this.discordMessage?.channel.send(i)
+                        this.discordMessage?.channel.send(userArray[value][i].quote)
+                        match = -1
+                      }
+                    }
+                    console.log(match)
+                    if (match == 0){
+                      this.discordMessage?.channel.send("The person has written no quote with the given phrase")
+                    }
+                  }
+                }
+                // :quote find
+                else if(this.findTypeSize == 1){
+                  if(this.generationType == "find"){
+                    for(let i = 0; i < Object.keys(userArray[value]).length; i = i + 1){
+                      this.discordMessage?.channel.send(i)
+                      this.discordMessage?.channel.send(userArray[value][i].quote)
+                    }
+                  }
+                }
+              }catch (error) {
+                console.log("Generator Type Find Is Not Working")
+              }
+            }
+            else if(this.generationType == "random" && Object.keys(userArray[value]).length > 0 ){
+              try{
+                var i = Math.floor(Math.random() * userArray[value].length);
+                this.discordMessage?.channel.send(i)
+                this.discordMessage?.channel.send(userArray[value][i].quote)
+                      //Will implement once more quotes are added.
+                      // else if(this.findType == "Random"){
+                      //   var a = Math.floor(Math.random() * userArray.length);
+                      //   var b = Math.floor(Math.random() * userArray[message.author.username].length);
+                      //   this.discordMessage?.channel.send(userArray[a].username)
+                      //   this.discordMessage?.channel.send(userArray[a][b].quote)
+                      // }
+              } catch(error){
+                console.log("Generator Type Random Is Not Working")
+              }
+            }
+            else if(this.generationType == "delete" && Object.keys(userArray[value]).length > 0 ){
+              try{
+                let number = this.findType[1]
+                if(this.findType[2] != null){
+                  number = number + this.findType[2]
+                }
+                if(this.findType[3] != null){
+                  number = number + this.findType[3]
+                }
+                if(parseInt(number)-1 <= Object.keys(userArray).length){
```

Comment:
If you do -1, then it should be < not <=
```suggestion
                if(parseInt(number)-1 < Object.keys(userArray).length){
```